src/main/java/org/apache/commons/collections4/IteratorUtils.java

```diff
@@ -907,6 +908,20 @@ public static NodeListIterator nodeListIterator(final Node node) {
         return new ZippingIterator<>(iterators);
     }
 
+    /**
+     * Returns an iterator that provides the elements contained in a pair of Iterators.
+     *
+     * @param <L> the left elements' type
+     * @param <R> the right elements' type
+     * @param left the iterator for the left side elements
+     * @param right the iterator for the right side elements
+     * @return an iterator, to iterate over the decorated iterators together until one is exhausted
+     * @throws NullPointerException if any iterator is null
```

Comment:
All new public and protected methods should have a Javadoc since 4.5 tag.

Reply:
Added a more descriptive Javadoc.